lib/macho/headers.rb

```diff
@@ -61,17 +69,69 @@ module MachO
   # @see http://llvm.org/docs/doxygen/html/Support_2MachO_8h_source.html
   CPU_SUBTYPE_LIB64 = 0x80000000
 
-  # all x86-type CPUs
+  # all CPU sub-types for CPU type `CPU_TYPE_I386`
   CPU_SUBTYPE_X86_ALL = 3
 
-  # all x86-type CPUs (what makes this different from CPU_SUBTYPE_X86_ALL?)
-  CPU_SUBTYPE_X86_ARCH1 = 4
+  # all CPU sub-types for CPU type `CPU_TYPE_X86_64`
+  CPU_SUBTYPE_X86_64_ALL = CPU_SUBTYPE_X86_ALL
+
+  # all CPU sub-types for CPU type `CPU_TYPE_X86_64` (Haskell tree)
+  CPU_SUBTYPE_X86_64_H = 8
+
+  # all CPU sub-types for CPU type `CPU_TYPE_ARM`
+  CPU_SUBTYPE_ARM_V7 = 9
+  CPU_SUBTYPE_ARM_V7S = 11
+  CPU_SUBTYPE_ARM_V7K = 12
+
+  # all CPU sub-types for CPU-type `CPU_TYPE_ARM64`
+  CPU_SUBTYPE_ARM64_ALL = 0
+
+  # PowerPC G3
+  CPU_SUBTYPE_POWERPC_750 = 9
 
-  # association of cpu subtypes to string representations
+  # PowerPC G4
+  CPU_SUBTYPE_POWERPC_7400 = 10
+
+  # PowerPC G4 "Voyager"
+  CPU_SUBTYPE_POWERPC_7450 = 11
+
+  # PowerPC G5
+  CPU_SUBTYPE_POWERPC_970 = 100
+
+  # all CPU sub-types for CPU type `CPU_TYPE_POWERPC`
+  CPU_SUBTYPE_POWERPC_ALL = 0
```

Comment:
The thing is that these always come in pairs and refer to certain architectures and instruction sets, not necessarily specific processors (of which some are capable of both 32-bit and 64-bit modes). Thus there's always one `CPU_TYPE_<type>` constant and at least one (but usually many) `CPU_SUBTYPE_<type>_<subtype>` constants strictly belonging to it. The sub-type field simply cannot be interpreted independently of the type field.